hayeon/programmers-2/모음-사전.js

```diff
@@ -0,0 +1,33 @@
+function solution(word) {
+  const result = [];
+  const str = "";
+
+  const dfs = (word, length, result) => {
+    const vowels = ["A", "E", "I", "O", "U"];
+
+    if (length === word.length) {
+      result.push(word);
+
+      return;
+    }
+
+    vowels.forEach((vowel) => {
+      dfs(word + vowel, length, result);
+    });
+  };
+
+  for (let i = 1; i <= 5; i++) {
+    dfs(str, i, result);
+  }
+
+  return result.sort().indexOf(word) + 1;
+}
```

Comment:
```suggestion
function solution(word) {
  const vowels = ["A", "E", "I", "O", "U"];
  const result = [];
  const str = "";

  const dfs = (word, length) => {
    if (word.length === length) {
      result.push(word);
      return;
    }

    vowels.forEach((vowel) => {
      dfs(word + vowel, length);
    });
  };

  for (let i = 1; i <= 5; i++) {
    dfs(str, i);
  }

  return result.sort().indexOf(word) + 1;
}
```
- 풀이잘봤습니다~!
- dfs의 인자인 result는 빼셔도 됩니다 어차피 전역에 있는 result에 push밖에 하지 않아서 result 인자는 필요하지 않습니다.
- vowels 배열은 전역으로 빼주셔도됩니다. dfs를 반복할 때마다 vowels를 매번 재 선언하는 것은 비효율적입니다.

Reply:
- 훨씬 이해하기도 쉽고 효율적으로 보입니다!!
- 어려워서 풀이를 약간 참고하고 풀었는데 result 인자를 굳이 안에 넣지 않아도 되었던거군요..
- 다시 한 번 풀어보겠습니다!! 감사합니다. 🙇‍♀️ 👍🏻